internal/notifications/notifications.go

```diff
@@ -0,0 +1,89 @@
+	"github.com/RHEnVision/provisioning-backend/internal/kafka"
+	"github.com/rs/zerolog"
+)
+
+func init() {
+	GetNotificationClient = getNoopNotificationClient
+}
+
+type client struct{}
+
+func getNotificationClient(ctx context.Context) NotificationClient {
+	zerolog.Ctx(ctx).Debug().Msg("Using kafka notification client")
+	return &client{}
+}
+
+func Initialize(ctx context.Context) {
+	zerolog.Ctx(ctx).Debug().Msg("Initialized kafka notification client")
+	GetNotificationClient = getNotificationClient
+}
+
+func (x *client) SuccessfulLaunch(ctx context.Context, reservationId int64) {
+	logger := zerolog.Ctx(ctx)
+	logger.Debug().Msg("Triggering a successful launch notification")
+	rDao := dao.GetReservationDao(ctx)
+	reservation, err := rDao.GetById(ctx, reservationId)
+	if err != nil {
+		logger.Warn().Err(err).Msg("unable to find reservation by id")
+		return
+	}
+	instances, err := rDao.ListInstances(ctx, reservationId)
+	if err != nil {
+		logger.Warn().Err(err).Msg("unable to get instances for reservation")
+	}
+	NotificationInstancesEvents := make([]kafka.NotificationEvent, len(instances))
+	for i, instance := range instances {
+		marshalInstance, er := json.Marshal(instance)
+		if er != nil {
+			logger.Warn().Err(err).Msg("unable to marshal instance")
+			return
+		}
+		NotificationInstancesEvents[i] = kafka.NotificationEvent{Payload: marshalInstance}
+	}
+
+	notificationMsg, err := kafka.NotificationMessage{Context: reservation, EventType: kafka.NotificationSuccessEventType, Events: NotificationInstancesEvents}.GenericMessage(ctx)
+	if err != nil {
+		logger.Warn().Err(err).Msg("unable to create notification message")
+		return
+	}
+	logger.Info().Msgf("Sending notification message")
+	err = kafka.Send(ctx, &notificationMsg)
+	if err != nil {
+		logger.Warn().Err(err).Msg("Unable to send notification message via kafka")
+	}
+}
+
+func (x *client) FailedLaunch(ctx context.Context, reservationId int64, jobError string) {
+	logger := zerolog.Ctx(ctx)
+	logger.Debug().Msg("Triggering a failed launch notification")
+	rDao := dao.GetReservationDao(ctx)
+	reservation, err := rDao.GetById(ctx, reservationId)
+	if err != nil {
+		logger.Warn().Err(err).Msg("unable to find reservation by id")
+		return
+	}
+	marshalError, err := json.Marshal(kafka.NotificationError{Error: jobError})
+	if err != nil {
+		logger.Warn().Err(err).Msg("unable to marshal error")
+		return
+	}
+
+	notificationEvent := []kafka.NotificationEvent{{Payload: marshalError}}
+	notificationMsg, err := kafka.NotificationMessage{Context: reservation, EventType: kafka.NotificationFailureEventType, Events: notificationEvent}.GenericMessage(ctx)
+	if err != nil {
+		logger.Warn().Err(err).Msg("unable to create notification failure message")
+		return
+	}
+	logger.Info().Msgf("Sending notification message")
+	err = kafka.Send(ctx, &notificationMsg)
+	if err != nil {
+		logger.Warn().Err(err).Msg("Unable to send notification message via kafka")
+	}
```

Comment:
This must be error log, because warnings do not propagate into Sentry. We want to know if Kafka is down via Sentry.